### PR TITLE
Some note touchups

### PR DESCRIPTION
**Pull request by cathieyun**
I realized that there are some parts in the write-up that have a gap in explanation / require a big jump in understanding, and I’d like to fill those in a little more.

Also, fix up the rangeproof doctests & add an example doctest to the README.

Also, add some of the changes that were made to the notes (on the circuit branch) in this PR:
https://github.com/dalek-cryptography/bulletproofs/pull/191/files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,14 +76,65 @@ This code is still research-quality.  It is not (yet) suitable for
 deployment.  The development roadmap can be found in the
 [Milestones][gh_milestones] section of the [Github repo][gh_repo].
 
-## Tests
+## Example
 
-Run tests with `cargo test`.
+```
```
Comment (oleganza) [inline README.md]: maybe add `rust` to enable the syntax highlighter?

    ```rust
    
    ```

File: README.md [hunk 2]
```diff
@@ -76,14 +76,65 @@ This code is still research-quality.  It is not (yet) suitable for
...
+# extern crate rand;
```
Comment (oleganza) [inline README.md]: Is this snippet in README really a doctest? if it's not, we can remove all pound-prefixed lines entirely.

Comment (hdevalence) [reply README.md]: Yes, it is: the README is included as a doc-str for the crate's root module, and this snippet is run as a doctest in that context.

Comment (cathieyun) [reply README.md]: it is a doctest! it runs with `cargo test --doc`

File: README.md [hunk 3]
```diff
@@ -76,14 +76,65 @@ This code is still research-quality.  It is not (yet) suitable for
...
+        .verify_single(&bp_gens, &pc_gens, &mut verifier_transcript, &committed_value, 32)
+        .is_ok()
+);
+# }
+```
 
-## Benchmarks
+## Tests and Benchmarks
+
+Run tests with `cargo test`.
```
Comment (hdevalence) [inline README.md]:
```suggestion
```
These could be in one paragraph

File: docs/notes.md [hunk 4]
```diff
@@ -270,30 +270,36 @@ we finally obtain
...
+the single inner-product equation (unblinded \\({\mathbf{l}(x)}\\) and \\({\mathbf{r}(x)}\\)) to the verifier without revealing information
 about the value \\(v\\), and since the inner-product argument is not
 zero-knowledge, they cannot be used there either.
 
 Instead, the prover chooses vectors of blinding factors
 \\[
 {\mathbf{s}}\_{L}, {\mathbf{s}}\_{R} \\;{\xleftarrow{\\$}}\\; {\mathbb Z\_p}^{n},
 \\]
-and uses them to construct vector polynomials
+and uses them to construct blinded vector polynomials from unblinded \\({\mathbf{l}(x)}\\) and \\({\mathbf{r}(x)}\\):
```
Comment (hdevalence) [inline docs/notes.md]:
```suggestion
and uses them to construct blinded vector polynomials from the unblinded vector polynomials \\({\mathbf{l}(x)}\\) and \\({\mathbf{r}(x)}\\):
```
small repetition for clarity

File: docs/notes.md [hunk 5]
```diff
@@ -606,7 +612,10 @@ additional and final step involves sending a pair of scalars
 Aggregated Range Proof
 ======================
 
-We want to take advantage of the logarithmic size of the inner-product protocol, by creating an aggregated range proof for \\(m\\) values that is smaller than \\(m\\) individual range proofs.
+The goal of an _aggregated range proof_ is to enable a group of parties to produce proofs of their individual statements
+(individual range proofs for the corresponding value commitments), that can be aggregated in a more compact proof.
+This is made efficient due to a logarithmic size of the inner-product protocol: an aggregated range proof for \\(m\\)
```
Comment (hdevalence) [inline docs/notes.md]:
```suggestion
This is more efficient due to the logarithmic size of the inner-product protocol: an aggregated range proof for \\(m\\)
```

File: src/range_proof/mod.rs [hunk 6]
```diff
@@ -104,23 +104,23 @@ impl RangeProof {
     ///
     /// // The proof can be chained to an existing transcript.
     /// // Here we create a transcript with a doctest domain separator.
-    /// let mut transcript = Transcript::new(b"doctest example");
+    /// let mut prover_transcript = Transcript::new(b"doctest example");
```
Comment (hdevalence) [inline src/range_proof/mod.rs]: :+1:

File: README.md (hunk 2 above)
Comment (oleganza) [reply README.md]: Cool!

**Pull request merged**